src/expandableCalendar/week.tsx

```diff
@@ -1,33 +1,51 @@
 import XDate from 'xdate';
 import React, {useRef, useMemo, useCallback} from 'react';
 import {View} from 'react-native';
+import isEqual from 'lodash/isEqual';
 
 import {getPartialWeekDates, getWeekDates, sameMonth} from '../dateutils';
 import {parseDate, toMarkingFormat} from '../interface';
 import {getState} from '../day-state-manager';
 import {extractDayProps} from '../componentUpdater';
 import styleConstructor from './style';
-import Calendar, {CalendarProps} from '../calendar';
+import {CalendarProps} from '../calendar';
 import Day from '../calendar/day/index';
-// import BasicDay from '../calendar/day/basic';
+import {CalendarContextProps} from './Context';
 
 
-export type WeekProps = CalendarProps;
+export type WeekProps = CalendarProps & {
+  context?: CalendarContextProps;
+};
+
+const propsAreEqual = (prevProps: WeekProps, nextProps: WeekProps) => {
+  const prevPropsToCheck = {
+    ...prevProps.markedDates,
+    ...prevProps.context,
+  };
+
+  const nextPropsToCheck = {
+    ...nextProps.markedDates,
+    ...nextProps.context,
+  };
 
-const Week = (props: WeekProps) => {
+  return isEqual(prevPropsToCheck, nextPropsToCheck);
```

Comment:
What if these props are equal but some other prop changed, like 'firstDay'? It will still return true and there won't be an update to the component.